### DISCUSSION TITLE
DeriveColumn error

### DISCUSSION THREAD

**Discussion by ringrid** [SeaQL/sea-orm#1279] [General]
version: sea-orm 0.10.5

I tried to use custom select

```rust
#[derive(Copy, Clone, Debug, EnumIter, DeriveColumn)]
enum QueryAs {
        CakeName,
}
```
but got some error for:  no method named `as_str` found for reference `&QueryAs` in the current scope

**Comment by billy1624** [comment]
Hey @ringrid, hi there. Could you upload a reproducible example to GitHub or pastebin?

**Comment by ringrid** [comment]
I found this enum had no error when together with entity, but i did not know the reason....

**Comment by ringrid** [comment]
entity.rs
```rust
use sea_orm::entity::prelude::*;

#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq)]
#[sea_orm(table_name = "pool")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub name: String,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
```
----------------------------------------------------------

main.rs
```rust
use sea_orm::{
    Database, DeriveColumn, EntityTrait, EnumIter, FromQueryResult, QuerySelect,
};

mod entity;

#[derive(Copy, Clone, Debug, EnumIter, DeriveColumn)]
enum QueryAs {
    PoolName,
}

#[derive(Debug, FromQueryResult)]
struct PoolResult {
    name: String,
}

#[tokio::main]
async fn main() {
    let db = Database::connect("xxxx").await.unwrap();

    let result1 = entity::Entity::find()
        .select_only()
        .column(entity::Column::Name)
        .into_model::<PoolResult>()
        .all(&db)
        .await
        .unwrap();

    let result2: Vec<String> = entity::Entity::find()
        .select_only()
        .column_as(entity::Column::Name, QueryAs::PoolName)
        .into_values::<_, QueryAs>()
        .all(&db)
        .await
        .unwrap();
}
```

DeriveColumn has error, but when I put entity.rs into main.rs，it has no error

**Reply by billy1624** [reply]
Hey @ringrid, I confirm this is a bug originated from the fact that `IdenStatic::as_str()` was used here. It assumed someone should import `sea_orm::IdenStatic` into scope where the `DeriveColumn` procedural macro is being used.

https://github.com/SeaQL/sea-orm/blob/ee2afc4b9103350df49557596027e25533ae1276/sea-orm-macros/src/derives/column.rs#L134-L139

I just fix the bug by refactoring the code, see https://github.com/SeaQL/sea-orm/pull/1280.

As a temporary patch. Try importing `IdenStatic` into scope, i.e. `use sea_orm::IdenStatic`.

**Reply by billy1624** [reply]
Thanks for the report!! Let me know if you still have question on that

**Comment by ringrid** [comment]
@billy1624 yeah, thanks a lot, it can work well.
